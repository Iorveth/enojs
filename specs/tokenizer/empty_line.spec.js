const tokenize = require('../../lib/parse_steps/tokenize.js');

let input = '\n' +
            ' \n' +
            '  \n' +
            '   \n' +
            '\n' +
            ' \n' +
            '  \n' +
            '   \n';

describe('Empty line tokenization', () => {
  it('performs to specification', () => {
    const context = { input: input };

    tokenize(context);

    expect(context.instructions).toMatchSnapshot();
  });
});
