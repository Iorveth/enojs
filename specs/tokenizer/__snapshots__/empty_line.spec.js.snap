// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty line tokenization performs to specification 1`] = `
Array [
  Object {
    "index": 0,
    "length": 0,
    "line": 0,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 1,
    "length": 1,
    "line": 1,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 3,
    "length": 2,
    "line": 2,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 6,
    "length": 3,
    "line": 3,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 10,
    "length": 0,
    "line": 4,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 11,
    "length": 1,
    "line": 5,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 13,
    "length": 2,
    "line": 6,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 16,
    "length": 3,
    "line": 7,
    "type": "EMPTY_LINE",
  },
  Object {
    "index": 20,
    "length": 0,
    "line": 8,
    "type": "EMPTY_LINE",
  },
]
`;
